### PR TITLE
feat(order): shard testing packages

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

The testing package has been split into `@daffodil/order/state/testing`, `@daffodil/order/driver/testing`, and `@daffodil/rder/driver/in-memory`.

## Other information

### CODE REVIEW

File: libs/order/driver/testing/package.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "name": "@daffodil/order/driver/magento/testing"
```
Comment (lderrickable) [inline libs/order/driver/testing/package.json:2]: this should be `@daffodil/order/driver/testing`

File: libs/order/state/src/selectors/order.selector.spec.ts [hunk 2]
```diff
@@ -3,17 +3,17 @@ import { Store, StoreModule, select, combineReducers } from '@ngrx/store';
 import { cold } from 'jasmine-marbles';
 
 import { DaffOrder } from '@daffodil/order';
+  import {
+    daffOrderReducers,
+    DaffOrderReducersState,
+    DAFF_ORDER_STORE_FEATURE_KEY,
+    DaffOrderListSuccess
+  } from '@daffodil/order/state';
```
Comment (lderrickable) [inline libs/order/state/src/selectors/order.selector.spec.ts:11]: indentation is too far in

File: libs/order/driver/testing/package.json (hunk 1 above)
Comment (griest024) [reply libs/order/driver/testing/package.json]: changed

File: libs/order/state/src/selectors/order.selector.spec.ts (hunk 2 above)
Comment (griest024) [reply libs/order/state/src/selectors/order.selector.spec.ts]: changed

**Review by lderrickable — APPROVED** [review]

**Pull request merged**